contracts/SwapAndBridge.sol

```diff
@@ -291,4 +328,178 @@ contract UniversalSwapAndBridge is SwapAndBridgeBase {
+     * Caller can specify their slippage tolerance for the swap and Across deposit params.
+     * @dev If swapToken does not implement `receiveWithAuthorization` to the specifications of EIP-3009, this call will revert.
+     * @param swapToken Address of the token that will be swapped for acrossInputToken.
+     * @param acrossInputToken Address of the token that will be bridged via Across as the inputToken.
+     * @param routerCalldata ABI encoded function data to call on router. Should form a swap of swapToken for
+     * enough of acrossInputToken, otherwise this function will revert.
+     * @param swapTokenAmount Amount of swapToken to swap for a minimum amount of depositData.inputToken.
+     * @param minExpectedInputTokenAmount Minimum amount of received depositData.inputToken that we'll submit bridge
+     * deposit with.
+     * @param depositData Specifies the Across deposit params we'll send after the swap.
+     * @param validAfter The unix time after which the `receiveWithAuthorization` signature is valid.
+     * @param validBefore The unix time before which the `receiveWithAuthorization` signature is valid.
+     * @param nonce Unique nonce used in the `receiveWithAuthorization` signature.
+     * @param v v of the EIP-3009 signature.
+     * @param r r of the EIP-3009 signature.
+     * @param s s of the EIP-3009 signature.
+     */
+    function swapAndBridgeWithAuthorization(
+        IERC20Auth swapToken,
+        IERC20 acrossInputToken,
+        bytes calldata routerCalldata,
+        uint256 swapTokenAmount,
+        uint256 minExpectedInputTokenAmount,
+        DepositData calldata depositData,
+        uint256 validAfter,
+        uint256 validBefore,
+        bytes32 nonce,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external nonReentrant {
+        // While any contract can vacuously implement `transferWithAuthorization` (or just have a fallback),
+        // if tokens were not sent to this contract, by this call to the swapToken, the call to `transferFrom`
+        // in _swapAndBridge will revert.
+        swapToken.receiveWithAuthorization(
+            msg.sender,
+            address(this),
+            swapTokenAmount,
+            validAfter,
+            validBefore,
+            nonce,
+            v,
+            r,
+            s
+        );
+        IERC20 _swapToken = IERC20(address(swapToken)); // Cast IERC20Auth to IERC20.
+
+        _swapAndBridge(
+            routerCalldata,
+            swapTokenAmount,
+            minExpectedInputTokenAmount,
+            depositData,
+            _swapToken,
+            acrossInputToken
+        );
+    }
+
+    /**
+     * @notice Deposits an EIP-2612 token Across input token into the Spoke Pool contract.
+     * @dev If `acrossInputToken` does not implement `permit` to the specifications of EIP-2612, this function will fail.
```

Comment:
Same q. RE the try/catch wrapper - is this comment still correct?

Reply:
You can think of this as: if a user uses a token which doesn't implement permit, then all else equal, this function will fail on `safeTransferFrom`, since `permit` does nothing to increase the approval. This is different from a token which does implement permit and has had their signature executed before the call to this function, since at least in that case, the allowance is expected to have changed.